src/main/java/com/minecolonies/coremod/entity/ai/citizen/cook/EntityAIWorkCook.java

```diff
@@ -304,8 +318,26 @@ protected int getActionsDoneUntilDumping()
     }
 
     @Override
-    protected IRequestable getSmeltAbleClass()
-    {
+    protected IRequestable getSmeltAbleClass() {
+        final Map<String, List<ItemStorage>> allowedItems = getOwnBuilding().getCopyOfAllowedItems();
+        if (allowedItems.containsKey(FOOD_LIST))
+        {
+            List<ItemStorage> list = ImmutableList.copyOf(IColonyManager.getInstance()
```

Comment:
We don't have to do the filter here. We can simply check, if the allowedItems.get(FOOD_LIST) is as big as the actual food list in the compat manager.